src/neuroconv/datainterfaces/behavior/deeplabcut/deeplabcutdatainterface.py

```diff
@@ -85,6 +316,122 @@ def get_metadata(self):
+        edges = []
+        try:
+            from ._dlc_utils import _get_graph_edges
+
+            filename = str(Path(file_path).parent / Path(file_path).stem)
+            for i, c in enumerate(filename[::-1]):
+                if c.isnumeric():
+                    break
+            if i > 0:
+                filename = filename[:-i]
+
+            metadata_file_path = Path(filename + "_meta.pickle")
+            edges = _get_graph_edges(metadata_file_path=metadata_file_path)
+        except Exception:
+            pass
+
+        # Create default PoseEstimation metadata
+        container_name = self.pose_estimation_metadata_key
+        skeleton_name = f"Skeleton{container_name}_{self.subject_name.capitalize()}"
+        device_name = f"Camera{container_name}"
+
+        # Create PoseEstimation metadata structure
+        pose_estimation_metadata = DeepDict()
+
+        # Add Skeleton as a dictionary
+        pose_estimation_metadata["Skeletons"] = {
+            skeleton_name: {"name": skeleton_name, "nodes": bodyparts, "edges": edges, "subject": self.subject_name}
+        }
+
+        # Add Device as a dictionary
+        pose_estimation_metadata["Devices"] = {
+            device_name: {
+                "name": device_name,
+                "description": "Camera used for behavioral recording and pose estimation.",
+            }
+        }
+
+        # Extract video name and scorer
+        video_name, scorer = Path(file_path).stem.split("DLC")
+        scorer = "DLC" + scorer
+
+        # Get video info from config file if available
+        video_file_path = None
+        image_shape = "0, 0, 0, 0"
+
+        if self.source_data.get("config_file_path"):
+            from ._dlc_utils import _get_video_info_from_config_file
+
+            video_file_path, image_shape = _get_video_info_from_config_file(
+                config_file_path=self.source_data["config_file_path"], vidname=video_name
+            )
+
+        # Add PoseEstimation container
+        pose_estimation_metadata["PoseEstimationContainers"] = {
+            container_name: {
+                "name": container_name,
+                "description": "2D keypoint coordinates estimated using DeepLabCut.",
+                "source_software": "DeepLabCut",
+                "dimensions": dimensions,
+                "skeleton": skeleton_name,
+                "devices": [device_name],
+                "scorer": scorer,
+                "original_videos": [video_file_path] if video_file_path else None,
+                "PoseEstimationSeries": {},
+            }
+        }
+
+        # Add a series for each bodypart
+        for bodypart in bodyparts:
+            pose_estimation_metadata["PoseEstimationContainers"][container_name]["PoseEstimationSeries"][bodypart] = {
+                "name": f"{self.subject_name}_{bodypart}",
+                "description": f"Keypoint {bodypart} from individual {self.subject_name}.",
```

Comment:
What do you think about leaving the subject_name out of these descriptions by default? Ex.
```python
"name": f"{bodypart}",
"description": "Keypoint for {bodypart}"
```

It would make my life a lot easier with fewer metadata fields to adjust when renaming subjects, and there's only one PoseEstimation container per subject anyway, so it would remain unambiguous.

Reply:
This makes sense to me.

Reply:
I would like to this and the subject modification in a follow up PR after this big change though. Would you be Ok with that?